rpc/shared/shared/src/main/scala/io/udash/rpc/serialization/ExceptionsRegistry.scala

```diff
@@ -0,0 +1,41 @@
+package io.udash.rpc.serialization
+
+import com.avsystem.commons.serialization.GenCodec
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+trait ExceptionsRegistry {
+  /** Registers GenCodec for provided class name. It can override previous value. */
+  def register[T: ClassTag](codec: GenCodec[T]): Unit
+  /** Returns GenCodec for provided class name. */
+  def get[T](name: String): GenCodec[T]
+  /** Returns true, if contains GenCodec for provided class name. */
+  def contains(name: String): Boolean
+  /** Returns name used for class T. */
+  def name[T](ex: T): String
+}
+
+object ExceptionsRegistry {
+  val Default: ExceptionsRegistry = new DefaultExceptionsRegistry
+
+  private class DefaultExceptionsRegistry extends ExceptionsRegistry {
+    import scala.reflect._
+
+    private val codecs: mutable.Map[String, GenCodec[_]] = mutable.Map.empty
```

Comment:
It might be nice to automatically register default codecs for standard library exceptions that are already available in both JVM and JS (e.g. `IllegalArgumentException` etc.)